### DISCUSSION TITLE
Songs delete/overwrite each other

### DISCUSSION THREAD

**Discussion by doginuu** [ppy/osu#26260] [Q&A] [answered]
### Type

Game behaviour

### Bug description

I have a lot of unsubmitted maps that are functional in stable but seemingly did not import at first. I've learned individually the song downloads and can be played fine however if I download other seemingly unrelated map(s) it deletes whichever one I had in my library to add the new one (if I add all of them together it chooses one out of the bunch to keep in the library).

This could be a pain as a mapper if you lost an unsubmitted WIP due to a strange overwrite phenomenon but currently I was trying to figure out why a collection I imported from stable with 827 maps appeared in lazer as 823 maps and it was because only 1/3 of the sets that overwrite each other can be present at a time.

First github post sorry I don't know how to format.

### Screenshots or videos

Demonstration:
https://github.com/ppy/osu/assets/155230998/83c6b91a-f96d-4338-9fb7-4f115c6bcb6f

.osz's:
[song 1 and 2.zip](https://github.com/ppy/osu/files/13797834/song.1.and.2.zip)
[song 3.zip](https://github.com/ppy/osu/files/13797835/song.3.zip)

### Version

Lazer 2023.1229.1

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/13797844/compressed-logs.zip)

**Comment by peppy** [comment]
These beatmaps contain completely fucked metadata. How did you make them? They are implying they are beatmap id 0 and set id 3, both of them. So to the game they are the same beatmap.

This can't happen unless you did this yourself, to my knowledge. I've never seen this before.

**Reply by doginuu** [reply]
elchxyrlia made them ages ago not me. Definitely from a much older version of stable and he uploaded them as a part of a massive ZIP for download. I'm not sure how he achieved this but this must not be an issue if it cannot be conventionally replicated. I've also noticed a lot of his unsubmitted maps in lazer will have the green "Ranked" text in lazer's song selection and if I click the webpage link it's a ranked map with the same song so he must've used them as templates.

They may have at one point been in the same set since he often uploads maps like this:
https://osu.ppy.sh/beatmapsets/338608#osu/749134

His metadata can be pretty bad, I got an unsubmitted map with a normal and sped up version and the sped up version mp3 wasn't there so I made it myself. Now I suppose I do "beatmapsetID:-1" to import them properly?

**Reply by doginuu** [reply, marked as answer]
"Now I suppose I do "beatmapsetID:-1" to import them properly?" I'll mark this as the answer as it works but I'm not sure if you had wanted to make lazer able to load every map stable can.

Stable-related sidenote: Every time I exported Snow Prism [Ar9.5] (NOT unnamed diff) it would reset the beatmapsetID from -1 to 3. I had to delete the .osu and create a new difficulty using the unnamed diff as a template and now it doesn't reset. Very odd, both the deleted .osu and the new one are identical but the old one continued to resave as beatmapsetID:3. I don't know what I'd do if it was a different difficulty entirely and I couldn't recreate it in 20 seconds with the other diff being identical.
